flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/DynamicTableFactory.java

```diff
@@ -93,23 +96,51 @@ public DynamicTableFactory(ArcticCatalog arcticCatalog,
     this.internalCatalogName = internalCatalogName;
   }
 
+  public DynamicTableFactory() {
+  }
+
+  /**
+   * If table is create by ddl 'connector' option, not catalog.
+   * e.g. CREATE TABLE t (XXX) WITH ('connector'='arctic', ...);
+   */
+  private void initCatalogInfo(Map<String, String> options) {
+    if (internalCatalogName != null && internalCatalogBuilder != null) {
+      return;
+    }
+    String metastoreUrl = options.get(METASTORE_URL);
+    Preconditions.checkNotNull(metastoreUrl, String.format("%s should be set", METASTORE_URL));
+    internalCatalogBuilder = InternalCatalogBuilder.builder().metastoreUrl(metastoreUrl);
```

Comment:
What if two arctic catalogs(data source) exist in one Flink SQL? These internalCatalogBuilders would be in conflict

Reply:
I have changed the global variable to local to avoid the scenario.